cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectS2IAssemblyOperator.java

```diff
@@ -102,12 +112,89 @@ public KafkaConnectS2IAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability
                     .compose(i -> imagesStreamOperations.reconcile(namespace, KafkaConnectS2IResources.targetImageStreamName(connect.getCluster()), connect.generateTargetImageStream()))
                     .compose(i -> podDisruptionBudgetOperator.reconcile(namespace, connect.getName(), connect.generatePodDisruptionBudget()))
                     .compose(i -> buildConfigOperations.reconcile(namespace, KafkaConnectS2IResources.buildConfigName(connect.getCluster()), connect.generateBuildConfig()))
-                    .compose(i -> deploymentConfigOperations.scaleUp(namespace, connect.getName(), connect.getReplicas()).map((Void) null));
+                    .compose(i -> deploymentConfigOperations.scaleUp(namespace, connect.getName(), connect.getReplicas()))
+                    .compose(i -> deploymentConfigOperations.readiness(namespace, connect.getName(), 1_000, this.operationTimeoutMs))
+                    .compose(i -> chainFuture.complete(), chainFuture)
+                    .setHandler(reconciliationResult -> {
+                        StatusUtils.setStatusConditionAndObservedGeneration(kafkaConnectS2I, kafkaConnectS2Istatus, reconciliationResult);
+                        kafkaConnectS2Istatus.setRestApiAddress(connect.getServiceName() + "." + namespace + ".svc:" + KafkaConnectS2ICluster.REST_API_PORT);
+                        kafkaConnectS2Istatus.setBuildConfigName(bc.getMetadata().getName());
```

Comment:
You should be able to use the `KafkaConnectS2IResources.buildConfigName(connect.getCluster())` here which will save you from storing the whole BuildConfig in the `bc` variable.